pkgs/development/tools/xcbuild/default.nix

```diff
@@ -39,5 +39,6 @@ in stdenv.mkDerivation rec {
     rmdir $out/usr
   '';
 
-  buildInputs = [ cmake zlib libxml2 libpng CoreServices CoreGraphics ImageIO ];
-}
\ No newline at end of file
+  buildInputs = [ cmake zlib libxml2 libpng ninja ]
+    ++ stdenv.lib.optionals stdenv.isDarwin [ CoreServices CoreGraphics ImageIO ];
```

Comment:
Well the "xcbuild.raw" builds fine. You can even make it build the wrapper but it won't be able to produce anything because of some mismatch with the clang flags.